Formula/pythran.rb

```diff
@@ -0,0 +1,116 @@
+  end
+
+  resource "gast" do
+    url "https://files.pythonhosted.org/packages/83/4a/07c7e59cef23fb147454663c3271c21da68ba2ab141427c20548ae5a8a4d/gast-0.4.0.tar.gz"
+    sha256 "40feb7b8b8434785585ab224d1568b857edb18297e5a3047f1ba012bc83b42c1"
+  end
+
+  resource "networkx" do
+    url "https://files.pythonhosted.org/packages/b0/21/adfbf6168631e28577e4af9eb9f26d75fe72b2bb1d33762a5f2c425e6c2a/networkx-2.5.1.tar.gz"
+    sha256 "109cd585cac41297f71103c3c42ac6ef7379f29788eb54cb751be5a663bb235a"
+  end
+
+  resource "ply" do
+    url "https://files.pythonhosted.org/packages/e5/69/882ee5c9d017149285cab114ebeab373308ef0f874fcdac9beb90e0ac4da/ply-3.11.tar.gz"
+    sha256 "00c7c1aaa88358b9c765b6d3000c6eec0ba42abca5351b095321aef446081da3"
+  end
+
+  def install
+    openblas = Formula["openblas"].opt_prefix
+    pythranrc = prefix/"config.pythranrc"
+    pythranrc.write <<~EOS
+      [compiler]
+      blas=openblas
+      include_dirs=#{openblas}/include
+      library_dirs=#{openblas}/lib
+    EOS
+
+    xy = Language::Python.major_minor_version "python3"
+    ENV.prepend_create_path "PYTHONPATH", libexec/"lib/python#{xy}/site-packages"
+
+    resources.each do |r|
+      r.stage do
+        system "python3", *Language::Python.setup_install_args(libexec)
+      end
+    end
+
+    system "python3", *Language::Python.setup_install_args(libexec)
+    bin.install Dir[libexec/"bin/*"]
+
+    bin.env_script_all_files(libexec/"bin",
+                             PYTHONPATH: ENV["PYTHONPATH"],
+                             PYTHRANRC:  "${PYTHRANRC:-#{pythranrc}}")
+  end
+
+  test do
+    pythran = Formula["pythran"].opt_bin/"pythran"
+    python = Formula["python@3.9"].opt_bin/"python3"
+    gcc_bin = Formula["gcc"].opt_bin
+    gcc_major_ver = Formula["gcc"].any_installed_version.major
+
+    (testpath/"dprod.py").write <<~EOS
+      #pythran export dprod(int list, int list)
+      def dprod(arr0, arr1):
+        return sum([x*y for x,y in zip(arr0, arr1)])
+    EOS
+    system pythran, testpath/"dprod.py"
+    rm_f testpath/"dprod.py"
+    assert_equal "11", shell_output("#{python} -c 'import dprod; print(dprod.dprod([1,2], [3,4]))'").chomp
+
+    (testpath/"arc_distance.py").write <<~EOS
+      #pythran export arc_distance(float[], float[], float[], float[])
+      import numpy as np
+      def arc_distance(theta_1, phi_1, theta_2, phi_2):
+        """
+        Calculates the pairwise arc distance between all points in vector a and b.
+        """
+        temp = np.sin((theta_2-theta_1)/2)**2 + np.cos(theta_1)*np.cos(theta_2)*np.sin((phi_2-phi_1)/2)**2
+        distance_matrix = 2 * np.arctan2(np.sqrt(temp), np.sqrt(1-temp))
+        return distance_matrix
+    EOS
+    with_env(
+      "CC"  => "#{gcc_bin}/gcc-#{gcc_major_ver}",
+      "CXX" => "#{gcc_bin}/g++-#{gcc_major_ver}",
+    ) do
+      system pythran, "-DUSE_XSIMD", "-fopenmp", "-march=native", testpath/"arc_distance.py"
```

Comment:
Not sure if GNU `gcc` should be set as default or not.
https://github.com/serge-sans-paille/pythran#mac-osx does mention potentially setting gcc.

This example is last one from https://pythran.readthedocs.io/en/latest/MANUAL.html#first-steps with recommended compile options, which are defined for GNU `gcc`.